contributor-docs/adrs/adr-XXX-file-structure.md

```diff
@@ -0,0 +1,37 @@
+# ADR XXX: File structure
+
+## Status
+
+Proposed
+
+## Context
+
+Components might be more grokable if they were structured consistently
+
+## Decision
+
+```
+primer-react/
+├─ src/
+│  ├─ BreadCrumbs/
+│  │  ├─ index.ts                    // Just re-exporting?
+│  │  ├─ BreadCrumbs.tsx             // Primary component
+│  │  ├─ BreadCrumbsItem.tsx         // Subcomponent
+│  │  ├─ BreadCrumbs.mdx             // Documentation. Always .mdx, not .md
+│  │  ├─ BreadCrumbs.stories.tsx
+│  │  ├─ BreadCrumbs.test.tsx
+│  │  └─ BreadCrumbs.types.test.tsx
+┆  ┆
+```
+
+Questions I would like to settle in this ADR:
+
+- On subcomponents:
+  - Do subcomponents filenames need to be namespaced with the parent component? (e.g., `BreadCrumbsItem.tsx` vs `Item.tsx`)
+  - Should the `index.ts` in each component directory export every subcomponent under its own name, or should we prefer `Object.assign`ing subcomponents to the parent component?
```

Comment:
Could it depend on whether the subcomponent is private to its parent and not intended for composition? I.e. With a compound component, I'd expect it the parent component to be the only one exported from the folder. Might be misunderstanding this point though.

Reply:
I think I've seen two different styles:
1. `BreadCrumbsItem`, which isn't exported on their own and is only accessed as `BreadCrumbs.Item`
2. `ButtonDanger`, which is available as its own export.
it is reasonable to me that those are different, as `BreadCrumbsItem` only makes sense as a child of `BreadCrumbs`.... and maybe we could name that as a rule? Alternately, we could consolidate to one style, right?